data-prepper-plugins/parse-json-processor/src/main/java/org/opensearch/dataprepper/plugins/codec/json/JsonOutputCodec.java

```diff
@@ -4,37 +4,71 @@
  */
 package org.opensearch.dataprepper.plugins.codec.json;
 
+import com.fasterxml.jackson.core.JsonEncoding;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
 import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
 import org.opensearch.dataprepper.model.codec.OutputCodec;
 import org.opensearch.dataprepper.model.event.Event;
 
 import java.io.IOException;
 import java.io.OutputStream;
+import java.util.Objects;
 
 /**
  * An implementation of {@link OutputCodec} which deserializes Data-Prepper events
  * and writes them to Output Stream as JSON Data
  */
-@DataPrepperPlugin(name = "json", pluginType = OutputCodec.class)
+@DataPrepperPlugin(name = "json", pluginType = OutputCodec.class, pluginConfigurationType = JsonOutputCodecConfig.class)
 public class JsonOutputCodec implements OutputCodec {
 
+    private JsonGenerator generator;
+    private static final String JSON = "json";
+    private static final JsonFactory factory = new JsonFactory();
+    JsonOutputCodecConfig config;
+
+    @DataPrepperPluginConstructor
+    public JsonOutputCodec(final JsonOutputCodecConfig config) {
+        Objects.requireNonNull(config);
+        this.config = config;
+    }
+
     @Override
     public void start(final OutputStream outputStream) throws IOException {
-        // TODO: do the initial wrapping like start the array
+        Objects.requireNonNull(outputStream);
+        generator = factory.createGenerator(outputStream, JsonEncoding.UTF8);
+        generator.writeStartArray();
     }
 
     @Override
-    public void writeEvent(final Event event, final OutputStream outputStream) throws IOException {
-        // TODO: get the event data and write event data to the outputstream
+    public void complete(final OutputStream outputStream) throws IOException {
+        generator.writeEndArray();
+        generator.close();
+        outputStream.flush();
+        outputStream.close();
     }
 
     @Override
-    public void complete(final OutputStream outputStream) throws IOException {
-        // TODO: do the final wrapping like closing outputstream and close generator
+    public void writeEvent(final Event event, final OutputStream outputStream) throws IOException {
```

Comment:
This implementation is not threadsafe. Because you are writing generator.writeStartObject and generator.writeEndObject as different calls, it's possible to write malformed json if multiple threads are running. You can either make this method synchronized or write the json into a byte array and then write into output stream

Reply:
The methods have been made synchronised

Reply:
start and complete were fine. writeEvent needs to be made synchronized.

Reply:
Okay, that has been taken care of.